### PR TITLE
Avoid access violations when a lib couldn't be loaded.

### PR DESCRIPTION
**Pull request by ace20022**

### CODE REVIEW

**Comment by Jalle19** [comment]
Is it really wise to hide these (potentially fatal) errors?

**Comment by ace20022** [comment]
Nothing gets hidden, and libcurl isn't required for many features of kodi.

File: xbmc/windowing/windows/WinSystemWin32.cpp [hunk 1]
```diff
@@ -444,6 +444,9 @@ bool CWinSystemWin32::ChangeResolution(RESOLUTION_INFO res)
 {
   const MONITOR_DETAILS &details = GetMonitor(res.iScreen);
 
+  if (!&details)
```
Comment (Paxxi) [inline xbmc/windowing/windows/WinSystemWin32.cpp]: this reads strange, as it's a reference it shouldn't be null and it's not a reference to a pointer either.

Comment (ace20022) [inline xbmc/windowing/windows/WinSystemWin32.cpp]: It can be wrong, GetMonitor returns m_MonitorsInfo[0] while m_MonitorsInfo is empty.

Comment (Paxxi) [inline xbmc/windowing/windows/WinSystemWin32.cpp]: Wrong yes but not null afaik so the check does nothing really.

Comment (Paxxi) [inline xbmc/windowing/windows/WinSystemWin32.cpp]: Also if GetMonitorInfo can return a bogus value it should return a pointer

Comment (ace20022) [inline xbmc/windowing/windows/WinSystemWin32.cpp]: on win32 it did what it should, but I had a bad feeling, I agree that it should return a pointer.

File: xbmc/filesystem/CurlFile.cpp [hunk 2]
```diff
@@ -940,6 +940,12 @@ void CCurlFile::Reset()
 
 bool CCurlFile::Open(const CURL& url)
 {
+  if (!g_curlInterface.IsLoaded())
+  {
+    CLog::Log(LOGDEBUG, "CurlFile::Open: curl interface not loaded");
```
Comment (wsnipex) [inline xbmc/filesystem/CurlFile.cpp]: This should be at the highest log level, not debug

Comment (ace20022) [inline xbmc/filesystem/CurlFile.cpp]: The highest is ```LOGFATAL``` which it isn't, is ```LOGERROR``` okay?

Comment (wsnipex) [inline xbmc/filesystem/CurlFile.cpp]: yeah, Error is ok then.

**Comment by ace20022** [comment]
@Paxxi Could you please have another look?

File: xbmc/windowing/windows/WinSystemWin32.cpp [hunk 3]
```diff
@@ -341,14 +341,24 @@ void CWinSystemWin32::RestoreDesktopResolution(int screen)
...
+const MONITOR_DETAILS* CWinSystemWin32::GetMonitor(int screen) const
 {
   for (unsigned int monitor = 0; monitor < m_MonitorsInfo.size(); monitor++)
     if (m_MonitorsInfo[monitor].ScreenNumber == screen)
-      return m_MonitorsInfo[monitor];
+      return &m_MonitorsInfo[monitor];
 
   // What to do if monitor is not found? Not sure... use the primary screen as a default value.
-  return m_MonitorsInfo[m_nPrimary];
+  if (m_nPrimary >= 0 && static_cast<unsigned int>(m_nPrimary) < m_MonitorsInfo.size())
```
Comment (Paxxi) [inline xbmc/windowing/windows/WinSystemWin32.cpp]: Should cast to size_t as it changes size on windows

File: xbmc/windowing/windows/WinSystemWin32.cpp [hunk 4]
```diff
@@ -363,12 +373,12 @@ int CWinSystemWin32::GetCurrentScreen()
...
 RECT CWinSystemWin32::ScreenRect(int screen)
 {
-  const MONITOR_DETAILS &details = GetMonitor(screen);
+  const MONITOR_DETAILS* details = GetMonitor(screen);
 
   DEVMODEW sDevMode;
   ZeroMemory(&sDevMode, sizeof(sDevMode));
   sDevMode.dmSize = sizeof(sDevMode);
-  if(!EnumDisplaySettingsW(details.DeviceNameW.c_str(), ENUM_CURRENT_SETTINGS, &sDevMode))
+  if(!EnumDisplaySettingsW(details->DeviceNameW.c_str(), ENUM_CURRENT_SETTINGS, &sDevMode))
```
Comment (Paxxi) [inline xbmc/windowing/windows/WinSystemWin32.cpp]: Should check details for nullptr before deref

**Comment by Paxxi** [comment]
Looks good apart from the minor comments. I agree with @ace20022 choice of log levels as the loss of curl is nonfatal

File: xbmc/windowing/windows/WinSystemWin32.cpp (hunk 4 above)
Comment (ace20022) [inline xbmc/windowing/windows/WinSystemWin32.cpp]: Replace all fault, thx!

**Comment by ace20022** [comment]
@Paxxi done, thx for the review!

**Pull request merged**